Sources/MapboxCoreNavigation/PassiveLocationDataSource.swift

```diff
@@ -22,16 +22,24 @@ open class PassiveLocationDataSource: NSObject {
         self.directions = directions
         
         let settingsProfile = SettingsProfile(application: ProfileApplication.kMobile, platform: ProfilePlatform.KIOS)
-        self.navigator = try! Navigator(profile: settingsProfile,
-                                        config: NavigatorConfig(),
-                                        customConfig: "",
-                                        tilesConfig: TilesConfig())
+        let config = try! ConfigFactory.build(for: settingsProfile, config: NavigatorConfig(), customConfig: "")
+        let runLoopExecutor = try! RunLoopExecutorFactory.build()
+        let historyRecorder = try! HistoryRecorderHandle.build(forConfig: config)
+        let cache = try! CacheFactory.build(for: TilesConfig(), config: config, runLoop: runLoopExecutor, historyRecorder: historyRecorder)
+        self.graphAccessor = try! GraphAccessor(cache: cache)
+        self.navigator = try! Navigator(config: config, runLoopExecutor: runLoopExecutor, cache: cache, historyRecorder: historyRecorder)
 
         self.systemLocationManager = systemLocationManager ?? NavigationLocationManager()
-        
+
         super.init()
-        
+
         self.systemLocationManager.delegate = self
+
+        try! self.navigator.setElectronicHorizonObserverFor(self)
```

Comment:
Should this happen unconditionally, or is there a use case for applications to benefit from map matching during free driving without the additional overhead of e-horizon updates?

Reply:
Well, at least it makes sense to subscribe to updates when the user sets the delegate. Adjusted the code